### PR TITLE
feat: Orchestration, Spring AI and Everything Else

### PR DESCRIPTION
**Pull request by MatKuhr**
## Context

Yes, context.

### Feature scope:
 
Yes.

## Definition of Done

- [ ] Functionality scope stated & covered
- [ ] Tests cover the scope above
- [ ] ~Error handling created / updated & covered by the tests above~
- [ ] ~Aligned changes with the JavaScript SDK~
- [ ] ~Documentation updated~
- [ ] ~Release notes updated~

### CODE REVIEW

File: orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationResponseHandler.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import org.apache.hc.core5.http.HttpEntity;
+import org.apache.hc.core5.http.ParseException;
+import org.apache.hc.core5.http.io.HttpClientResponseHandler;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+
+@Slf4j
+@RequiredArgsConstructor
+class OrchestrationResponseHandler<T> implements HttpClientResponseHandler<T> {
+  // TODO reduce code duplication with OpenAI
+  @Nonnull private final Class<T> responseType;
```
Comment (CharlesDuboisSAP) [inline orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationResponseHandler.java:24]: This is 99% code duplication with `OpenAIResponseHandler`

File: sample-code/spring-app/src/test/java/com/sap/ai/sdk/app/controllers/OrchestrationTest.java [hunk 2]
```diff
@@ -43,18 +41,6 @@ void template() {
...
-  void looseFilter() {
-    assertThat(new OrchestrationController().filter("4")).isNotNull();
-  }
-
-  @Test
-  void strictFilter() {
-    assertThatThrownBy(() -> new OrchestrationController().filter("0"))
-        .isInstanceOf(HttpClientErrorException.class)
-        .hasMessageContaining("400 Bad Request");
-  }
```
Comment (CharlesDuboisSAP) [inline sample-code/spring-app/src/test/java/com/sap/ai/sdk/app/controllers/OrchestrationTest.java:56]: Why remove my tests?

**Review by CharlesDuboisSAP — COMMENTED** [review]
Please split this PR in 2: first Orchestration Client and then Spring AI.
I did my best to review this but it's very early.

**Comment by MatKuhr** [comment]
will be split up into separate PRs. See #114

**Pull request closed without merging**